docs/ci-services.md

```diff
@@ -0,0 +1,55 @@
+# CI Services
+
+The CI service blob defines the supported CI services the user will be
+presented with when running the scaffolder.
+
+## Example
+
+```javascript
+{
+  TravisCI: {scaffolder: travisScaffolder, public: true}
+}
+```
+
+## Config
+
+### keys: __string__
+
+Keys in the CI services blob define the options the user will be presented with
+when choosing their desired CI service.
+
+example: `TravisCI`
+
+### values: __object__
+
+Values in the CI services blob define the actual mechanisms and configuration
+that will be used by the JavaScript Scaffolder to scaffold the CI service for
+the user's project.
+
+#### scaffolder: __function__ (_required_)
+
+Consumer-defined plugin(??) called when this ci service is selected by the
+user in the CLI. Will be called with the following arguments:
+
+* `projectRoot` __string__ full path to directory where scaffolder was initiated
+* `vcs` __object__
```

Comment:
this is a case where this shape is repeated/comes from other pieces of the scaffolding puzzle. i think spelling it out like this makes sense, but i do wonder if there would be some benefit at some point to have a shared shape defined somewhere that we link to from here (and other places) instead

Reply:
i very much agree here. at first i had empty links on each one of these to call that out because i do think more detail would be really useful, but obviously had no where to link them to yet 😄 . i can actually start that as a followup to this one, then link these to the new docs once they're there.

Reply:
> at some point

i'm not trying to say there is any rush to it, just calling out the thought that went through my head. i think this is very appropriate at this point, but seems like something we'll figure out as we get farther along with the remaining pieces